src/test/bufferlist.cc

```diff
@@ -2320,6 +2320,25 @@ TEST(BufferList, c_str) {
   EXPECT_EQ(0, ::memcmp("AB", bl.c_str(), 2));
 }
 
+TEST(BufferList, c_str_carriage) {
+  // verify the c_str() optimization for carriage handling
+  buffer::ptr bp("A", 1U);
+  bufferlist bl;
+  bl.append(bp);
+  bl.append('B');
+  EXPECT_EQ(2U, bl.get_num_buffers());
+  EXPECT_EQ(2U, bl.length());
+
+  // this should leave an empty bptr for carriage at the end of the bl
+  bl.splice(1, 1);
```

Comment:
Nit: I think it might be more indicative to have a single-buffer bufferlist and observe/expect `splice` turn it into a two-buffer bufferlist as this is how this came up.